harvesters/src/main/java/org/fao/geonet/kernel/harvest/harvester/csw/Aligner.java

```diff
@@ -631,4 +624,43 @@ private Element processMetadata(ServiceContext context,
         }
         return md;
     }
+
+    /**
+     * Retrieves the list of metadata uuids that have the same dataset identifier.
+     *
+     * @param searchMan         ES search manager.
+     * @param datasetIdCode     Dataset identifier.
+     * @return  A list of metadata uuids that have the same dataset identifier.
+     */
+    private Set<String> retrieveMetadataUuidsFromIdentifier(EsSearchManager searchMan,
+                                                            String datasetIdCode) {
+
+        Set<String> metadataUuids = new HashSet<>();
+
+        String jsonQuery = "{" +
+            "          \"term\": {" +
+            "            \"resourceIdentifier.code\": {" +
+            "              \"value\": \"%s\"" +
+            "            }" +
+            "          }" +
+            "}";
+        ObjectMapper objectMapper = new ObjectMapper();
+        try {
+            JsonNode esJsonQuery = objectMapper.readTree(String.format(jsonQuery, datasetIdCode));
+
+            final SearchResponse queryResult = searchMan.query(
+                esJsonQuery,
+                FIELDLIST_CORE,
+                0, 1000);
```

Comment:
Maybe only request the UUID

```suggestion
                FIELDLIST_UUID,
                0, 1000);
```

or even use the `hit.getId()`.

Also maybe we can search directly for `+resourceIdentifier.code:currentHarvestedRecordResourceIdentifier -uuid:currentHarvestedRecordUuid` ?